# 📦 Importação de Bibliotecas

Até agora, utilizamos apenas o Python “puro”, e isso está longe de ser um problema. A linguagem, por si só, já consegue realizar operações matemáticas, criar listas, manipular dados e tomar decisões com estruturas como if e else.

No entanto, conforme os problemas se tornam maiores e mais complexos, surge uma questão importante: vale a pena desenvolver tudo do zero?

Imagine, por exemplo, a necessidade de calcular o desvio padrão de 10.000 amostras ou ler uma planilha de Excel com vários gigabytes de dados. Embora fosse possível implementar essas soluções manualmente, isso exigiria muito tempo, aumentaria a complexidade do código e ainda abriria espaço para erros.

É nesse contexto que entram as bibliotecas.

Uma biblioteca é um conjunto de funções, métodos e classes criado para resolver problemas específicos. Existem bibliotecas voltadas para análise de dados, manipulação de arquivos, criação de gráficos, desenvolvimento web, bancos de dados, inteligência artificial e diversas outras áreas. Em geral, elas são desenvolvidas e mantidas por comunidades de programadores e disponibilizadas para uso público.

Na prática, utilizar bibliotecas significa aproveitar soluções que já foram testadas, otimizadas e amplamente utilizadas. Isso permite que o desenvolvedor concentre seus esforços na resolução do problema principal, em vez de gastar tempo implementando funcionalidades básicas do zero.

## 📤 A Sintaxe da Importação

Para utilizar uma biblioteca em Python, primeiro precisamos “trazê-la” para o nosso ambiente de trabalho. Esse processo é chamado de importação.
O Python possui três formas principais de fazer isso:


### 1. `import biblioteca`

Essa forma importa a biblioteca inteira para o programa.
Se você precisa calcular a raiz quadrada de um número, por exemplo, não precisa criar a lógica matemática. Basta importar o módulo nativo math:


In [2]:
# importamos toda a biblioteca `math` 
import math

#utilizamos a função `sqrt()` para calcular a raiz quadrada.
resultado = math.sqrt(25)
print(resultado)

5.0


### 2. `import biblioteca as apelido`

Aqui, também importamos a biblioteca inteira, mas atribuímos um apelido mais curto para facilitar a escrita do código.

Isso funciona de maneira semelhante às siglas utilizadas na área da saúde, como “PA” para Pressão Arterial ou “FC” para Frequência Cardíaca.


In [9]:
import math as m

# Uso com o apelido
raiz = m.sqrt(36)
print(raiz)

6.0


In [15]:
import pandas as pd

print("Importando os dados da pasta 'data'...\n")

# A função read_csv transforma o arquivo de texto em uma tabela inteligente (DataFrame)
dados_triagem = pd.read_csv("../data/teste01.csv")

# Exibindo a tabela na tela
display(dados_triagem)

ModuleNotFoundError: No module named 'pandas'

### 3. `from biblioteca import ferramenta`

Nessa abordagem, importamos apenas uma ferramenta específica da biblioteca.

Em vez de chamar o laboratório inteiro, solicitamos apenas o equipamento necessário para aquele exame.


In [5]:
# Importa apenas a função sqrt
from math import sqrt

# Calcula a raiz quadrada de 25
resultado = sqrt(25)

# Exibe o resultado
print(resultado)

5.0


In [6]:
# Importa apenas a função pow
from math import pow

# Calcula 2 elevado a 3
resultado = pow(2, 3)

print(resultado)

8.0


In [7]:
# Importa apenas randint
from random import randint

# Gera número aleatório
numero = randint(1, 10)

print(numero)

6


## Comparando as Abordagens

| Sintaxe                             | O que acontece                                 | Analogia                                       |
| ----------------------------------- | ---------------------------------------------- | ---------------------------------------------- |
| `import biblioteca`                 | Importa toda a biblioteca                      | Chamar o laboratório inteiro                   |
| `import biblioteca as apelido`      | Importa toda a biblioteca com um nome reduzido | Usar siglas em prontuários                     |
| `from biblioteca import ferramenta` | Importa apenas uma função específica           | Solicitar apenas um equipamento do laboratório |

---

## ⚡ A Importância da Velocidade: O Problema da Escala

Se você está analisando a idade de 10 pacientes no seu TCC, o Python tradicional (usando Listas) resolve o problema em um piscar de olhos. Mas e se o seu projeto de for analisar um banco de dados de sequenciamento genômico com **3 bilhões de pares de bases**? Ou processar imagens de ressonância magnética de alta resolução?

Se usarmos as Listas tradicionais do Python para dados massivos, o seu computador pode levar dias processando ou simplesmente pegar fogo, por falta de memória. É para isso que o **NumPy** existe: ele é o "motor turbo" projetado especificamente para a escala científica.

---

## 📦 Lista do Python vs. Array do NumPy: A Metáfora do Fichário

Por que o NumPy é tão mais rápido? A resposta está na forma como ele guarda a informação na memória do computador.

* **A Lista do Python (O Fichário Bagunçado):** Pode guardar qualquer coisa misturada (um texto, um número, um arquivo). Para achar e somar valores, o computador precisa "pular" de gaveta em gaveta na memória procurando os dados.
* **O Array do NumPy (A Caixa de Amostras Perfeita):** Só guarda dados do **mesmo tipo** (ex: só números inteiros). Ele organiza tudo em um bloco único e contíguo de memória. O computador lê de uma vez só, sem perder tempo procurando.

O NumPy é rápido porque organiza os dados de forma contínua na memória, permitindo processamento matemático eficiente em larga escala.


<br>

<img src="../assets/aula05_pythonList_numpyArray.png" width="800" alt="Comparação visual entre Listas Python e Arrays NumPy na memória">

<br>
  
---



In [1]:
import time
import numpy as np

tamanho_amostra = 10_000_000

print(f"Iniciando o processamento de {tamanho_amostra} amostras...\n")

# ---------------------------------------------------------
# TESTE 1: Python Tradicional (Listas)
# ---------------------------------------------------------
inicio_python = time.time() # Dispara o cronômetro

# Criando uma lista e somando (usando a função nativa sum)
lista_python = list(range(tamanho_amostra))
soma_python = sum(lista_python)

fim_python = time.time() # Para o cronômetro
tempo_python = fim_python - inicio_python


# ---------------------------------------------------------
# TESTE 2: O Especialista NumPy (Arrays)
# ---------------------------------------------------------
inicio_numpy = time.time() # Dispara o cronômetro

# Criando um array contíguo e somando
array_numpy = np.arange(tamanho_amostra)
soma_numpy = np.sum(array_numpy)

fim_numpy = time.time() # Para o cronômetro
tempo_numpy = fim_numpy - inicio_numpy

# ---------------------------------------------------------
# RESULTADOS DO EXPERIMENTO
# ---------------------------------------------------------
print("=== RESULTADOS ===")
print(f"Tempo do Python Puro: {tempo_python:.4f} segundos")
print(f"Tempo do NumPy:       {tempo_numpy:.4f} segundos")

# Calculando quantas vezes o NumPy foi mais rápido
vantagem = tempo_python / tempo_numpy
print(f"\n🚀 O NumPy foi aproximadamente {vantagem:.1f} VEZES mais rápido!")

Iniciando o processamento de 10000000 amostras...

=== RESULTADOS ===
Tempo do Python Puro: 0.3019 segundos
Tempo do NumPy:       0.0208 segundos

🚀 O NumPy foi aproximadamente 14.5 VEZES mais rápido!


# 📦 Arrays Multidimensionais

A principal estrutura do NumPy é o array.

### 🧪 O que é um Array? 

Um **array** é um tipo especial de variável utilizado para armazenar uma coleção de dados, ou seja, guardar múltiplas informações de uma só vez usando um único nome.

Para entender, faça uma analogia visual:
Imagine que você precisa transportar 10 amostras de sangue para a centrífuga. Você não carrega os tubos soltos, um em cada bolso, certo? Você os coloca em uma **estante para tubos de ensaio** (uma grade).

O Array é exatamente essa estante matemática. Em vez de criar 10 variáveis diferentes (`amostra1`, `amostra2`, `amostra3`...), você cria um único Array chamado `amostras_sangue` que guarda todas as 10 posições em ordem.

**A Regra de Ouro do Contexto (A Homogeneidade):**
Como você bem pensou, o ideal é que as informações em um array façam parte de um mesmo contexto. No NumPy, essa regra é levada ao extremo: **um array só aceita dados do mesmo tipo**.
Você não mistura um tubo de ensaio com uma seringa e um bisturi na mesma estante. Da mesma forma, um array de NumPy geralmente guarda *apenas* um mesmo tipo.

In [2]:
import numpy as np

In [2]:
# Sem usar Array (variáveis soltas - difícil de gerenciar)
pressao_paciente1 = 120
pressao_paciente2 = 125
pressao_paciente3 = 118
pressao_paciente4 = 130
pressao_paciente5 = 122
pressao_paciente6 = 11

# Usando um Array do NumPy (a nossa 'estante' organizada)
# Passamos uma lista de pressões sistólicas para a função np.array()
todas_pressoes = np.array([120, 125, 118, 130, 122, 119])

print("Amostras na estante:", todas_pressoes)
print("Quantas amostras temos?", len(todas_pressoes))

Amostras na estante: [120 125 118 130 122 119]
Quantas amostras temos? 6


## 📏 As Dimensões do Dado:

O NumPy organiza os dados em dimensões. Em matemática, chamamos isso de vetores, matrizes e tensores. Mas no nosso laboratório digital, podemos visualizar isso de uma forma muito mais concreta:

**1 Dimensão (Vetor):** É uma linha única de dados.
* *No Laboratório:* Pense em uma única fileira de tubos de ensaio.
* *Nos Dados:* Uma lista com a idade de 10 pacientes ou as aferições de pressão de um único paciente ao longo do dia.


**2 Dimensões (Matriz):** É uma tabela com linhas e colunas (altura e largura).
* *No Laboratório:* Pense em uma **Placa de Microtitulação (ELISA)** de 96 poços, ou uma planilha do Excel.
* *Nos Dados:* Linhas representando pacientes diferentes e Colunas representando exames diferentes (Idade, Peso, Glicemia).


**3 Dimensões ou mais (Tensor):** É um cubo de dados (profundidade, altura e largura).
* *No Hospital:* Pense em uma **Tomografia Computadorizada (TC)** ou Ressonância Magnética. O exame não é uma única imagem 2D (como um Raio-X), mas sim uma "pilha" de várias fatias 2D empilhadas para formar um volume 3D do órgão.

<br>

<img src="../assets/aula05_dimensoes_dado.jpeg" width="800" alt="As Dimensões do Dado">

<br>
  
---


## 🗺️ O Mapa do NumPy

Para dominar o **NumPy**, você não precisa decorar a documentação inteira. Se o Python padrão fosse uma calculadora de mão, o NumPy seria um supercomputador projetado para processar milhares de amostras clínicas ou sequências genômicas em frações de segundo.

Vamos focar nos **4 pilares fundamentais** dessa ferramenta.

## 1. A Estrutura Central: O `ndarray` (Array)

O coração do NumPy é o **Array**. Ao contrário das Listas nativas do Python (que aceitam uma bagunça de textos e números misturados), um Array exige **homogeneidade**: todos os dados devem ser do mesmo tipo. Essa rigidez é o que permite o processamento ultrarrápido.


In [4]:
# Cria um array NumPy com temperaturas corporais
# np.array() transforma uma lista Python em um array otimizado para cálculos numéricos
temperaturas = np.array([
    36.5,
    37.2,
    38.1,
    39.0,
    36.8
])


In [5]:
# .ndim => mostra o número de dimensões do array
# Resultado: 1
# Significa que temos um vetor (uma única sequência de dados)
print("Dimensões:", temperaturas.ndim)

Dimensões: 1


In [6]:
# .shape => mostra o formato estrutural do array
# Resultado: (5,)
# Significa:
# - 5 elementos
# - em uma única dimensão
print("Formato:", temperaturas.shape)


Formato: (5,)


In [7]:
# .dtype => mostra o tipo dos dados armazenados
# Resultado: float64
# Significa:
# - números decimais (float)
# - com alta precisão matemática
print("Tipo do dado:", temperaturas.dtype)

Tipo do dado: float64


## 🧠 Resumo Didático

| Método   | O que faz               | Resultado |
| -------- | ----------------------- | --------- |
| `.ndim`  | quantidade de dimensões | `1`       |
| `.shape` | formato do array        | `(5,)`    |
| `.dtype` | tipo dos dados          | `float64` |



<br>

<img src="../assets/aula05_resumo_didatico.jpeg" width="800" alt="Resumo Didático">

<br>

## 2. Vetorização: O Fim do Trabalho Braçal

Em Python puro, se você quiser multiplicar 1.000 amostras por 2, precisará usar um laço de repetição (`for`) e processar tubo por tubo. No NumPy, a matemática acontece em bloco, atingindo **todos os elementos ao mesmo tempo**.


In [6]:
# -----------------------------------
# CONVERSÃO DE CELSIUS → FAHRENHEIT
# -----------------------------------
# temperaturas = np.array([
#     36.5,
#     37.2,
#     38.1,
#     39.0,
#     36.8
# ])
# Fórmula matemática da conversão:
# Fahrenheit = (Celsius * 1.8) + 32

# O NumPy aplica a operação em TODOS os elementos automaticamente
# Isso é chamado de:
# vetorização (vectorization)

temp_fahrenheit = (
    temperaturas * 1.8
) + 32


# Exibe o novo array convertido
print(
    "Temperaturas em Fahrenheit:",
    temp_fahrenheit
)

Temperaturas em Fahrenheit: [ 97.7   98.96 100.58 102.2   98.24]


# 🔬 O que aconteceu?

O NumPy realizou:

| Celsius | Fahrenheit |
| ------- | ---------- |
| 36.5°C  | 97.7°F     |
| 37.2°C  | 98.96°F    |
| 38.1°C  | 100.58°F   |
| 39.0°C  | 102.2°F    |
| 36.8°C  | 98.24°F    |


sem usar:

```
for
```

---

# ⚡ Conceito Principal

O NumPy aplica cálculos:

```
multiplicou e somou em todos os elementos simultaneamente
```

Isso torna o processamento:

* mais rápido
* mais limpo
* mais eficiente

Isso é chamado de:

```
vetorização
```

---


# 📊 Resumo Didático

| Operação     | O que faz                     |
| ------------ | ----------------------------- |
| `np.array()` | cria array NumPy              |
| `* 9/5`      | multiplica todos os valores   |
| `+ 32`       | soma 32 em todos os elementos |
| Vetorização  | cálculo simultâneo no array   |



<br>

<img src="../assets/aula05_recumo_didatico2.png" width="800" alt="Resumo Didático">

<br>



## 🎭 3. Máscaras Booleanas: O Filtro de Triagem

Como você encontra rapidamente os pacientes em risco dentro de um banco de dados gigantesco? 
Criando uma "máscara" (uma condição lógica) e aplicando sobre o array para extrair apenas o que importa.

Em termos simples, uma máscara booleana é uma pergunta lógica que você faz para o seu Array. O NumPy visita cada elemento (paciente) e responde individualmente com True (Verdadeiro, atende ao critério) ou False (Falso, não atende).

O processo acontece em 3 passos cirúrgicos:

* Os Dados Brutos: O array original com as medições.

* A Máscara (A Triagem): O teste lógico que gera um array invisível de True e False.

* A Extração: O NumPy usa essa "peneira" para deixar passar apenas os dados que receberam a etiqueta True.




In [9]:
# 1. Dados Brutos: Glicemia em jejum de 6 pacientes (mg/dL)
# Array com valores de glicemia em jejum (mg/dL)
# Cada número representa um paciente
glicemia = np.array([
    90,
    105,
    126,
    85,
    140,
    99
])

print(
    "1. Glicemia dos pacientes:",
    glicemia
)

print("-" * 50)

1. Glicemia dos pacientes: [ 90 105 126  85 140  99]
--------------------------------------------------


In [10]:
# 2. A Máscara: Quem tem critério laboratorial para Diabetes (Glicemia >= 126)?
# O operador >= pergunta:
# "O valor é maior ou igual a 126?"

# O NumPy aplica essa comparação
# em TODOS os elementos do array ao mesmo tempo
# (vetorização)

mascara_diabetes = glicemia >= 126

# Resultado:
# True  → paciente com glicemia crítica
# False → paciente fora do critério
print(
    "2. Máscara Booleana:",
    mascara_diabetes
)

print("-" * 50)

2. Máscara Booleana: [False False  True False  True False]
--------------------------------------------------


In [11]:
# 3. A máscara é usada dentro dos colchetes []
# para extrair apenas os valores True

pacientes_criticos = glicemia[mascara_diabetes]


# Resultado:
# apenas pacientes com glicemia >= 126

print(
    "3. Pacientes críticos:",
    pacientes_criticos
)

3. Pacientes críticos: [126 140]


# ⚡ Conceito Principal

O NumPy fez isto automaticamente:

```
90  >= 126 → False
105 >= 126 → False
126 >= 126 → True
85  >= 126 → False
140 >= 126 → True
99  >= 126 → False
```

sem precisar usar:

```
for
```

---

# 🧪 Interpretação Biomédica

Critério aplicado:

Glicemia **126 mg/dL**

Pacientes identificados com possível critério laboratorial para diabetes:

```
126 mg/dL
140 mg/dL
```

---

# 📊 Resumo Didático

| Conceito         | O que faz                   |
| ---------------- | --------------------------- |
| `>= 126`         | cria condição clínica       |
| Máscara booleana | marca True/False            |
| `array[mascara]` | filtra os dados             |
| Vetorização      | compara tudo ao mesmo tempo |


<br>

<img src="../assets/aula05_recumo_didatico3.png" width="800" alt="Resumo Didático 3">

<br>



## 🧮 4. Funções Universais

Na pesquisa científica, existem duas necessidades constantes:

1. **Processar estatísticas básicas instantaneamente** sobre milhares de medidas (médias, desvios padrões, picos).
2. **Criar dados sintéticos (Mock Data)** para testar se os nossos scripts e equações matemáticas estão funcionando antes de colocarmos as mãos em dados reais e sensíveis de pacientes (protegidos pela LGPD).

O NumPy resolve ambos os problemas de forma integrada. Vamos entender como ele faz isso sem que você precise reescrever fórmulas matemáticas do zero.

---

### 📊 Funções Universais e Estatística
Agora que temos nossa população ou paciente simulado, precisamos extrair os laudos estatísticos. 
No Python puro, para calcular uma média, você precisaria somar todos os itens usando um loop e depois dividir pelo total.
No NumPy, as funções estatísticas são Universais. Elas olham para o array como um corpo único e extraem o diagnóstico matemático em microsegundos.

### 🧰 Guia de Referência: Funções Estatísticas e `uFuncs`

No NumPy, as ferramentas matemáticas são divididas pela forma como atuam sobre a nossa "estante de amostras" (o Array):

1. **Funções de Agregação (Estatísticas):** Elas olham para o conjunto inteiro de dados e o reduzem a um único valor resumido (ex: calcular a média de 1.000 pacientes).
2. **Funções Universais (`uFuncs`):** Elas realizam operações matemáticas **elemento a elemento**, mantendo o tamanho original do array (ex: extrair a raiz quadrada do resultado de cada paciente).

| Função | O que faz? | Aplicação Prática na Saúde |
|---|---|---|
| `np.mean(array)` | Média aritmética | Encontrar a média de idade ou de pressão arterial da coorte |
| `np.median(array)` | Mediana (ponto central) | Útil quando há valores extremos (outliers) que distorcem a média |
| `np.std(array)` | Desvio padrão | Medir a variabilidade biológica (ex: oscilação da glicemia) |
| `np.var(array)` | Variância | Análise de dispersão dos dados laboratoriais |
| `np.min(array)` / `np.max(array)` | Valor mínimo / máximo | Identificar os picos críticos (ex: menor e maior frequência cardíaca) |
| `np.sum(array)` | Soma total | Somar calorias ingeridas ou total de células contadas |
| `np.percentile(array, q)` | Percentil específico | Gráficos de crescimento infantil ou faixas de corte de exames (ex: percentil 95) |

---

In [18]:
# Array com valores de pressão sistólica (mmHg)
# Cada valor representa a pressão de um paciente
pressao_sistolica = np.array([
    120,
    130,
    115,
    140,
    125,
    180
])
print(pressao_sistolica )

[120 130 115 140 125 180]


In [21]:
# -----------------------------------
# ANÁLISE ESTATÍSTICA COMPLETA
# -----------------------------------

# np.mean()
# Calcula a média aritmética dos valores
# Soma todos os elementos e divide pela quantidade
print(
    "Média da pressão:",
    np.mean(pressao_sistolica)
)


# np.median()
# Retorna a mediana (valor central)
# Muito útil quando existem valores extremos (outliers)
print(
    "Mediana:",
    np.median(pressao_sistolica)
)


# np.std()
# Calcula o desvio padrão
# Mede o quanto os dados variam em relação à média
print(
    "Desvio padrão:",
    np.std(pressao_sistolica)
)


# np.var()
# Calcula a variância
# Representa a dispersão matemática dos dados
print(
    "Variância:",
    np.var(pressao_sistolica)
)


# np.min()
# Retorna o menor valor do array
# Aqui identifica a menor pressão registrada
print(
    "Menor pressão:",
    np.min(pressao_sistolica)
)


# np.max()
# Retorna o maior valor do array
# Aqui identifica um possível pico hipertensivo
print(
    "Maior pressão:",
    np.max(pressao_sistolica)
)


# np.sum()
# Soma todos os valores do array
# Pode representar totais acumulados
print(
    "Soma total das pressões:",
    np.sum(pressao_sistolica)
)


# np.percentile()
# Calcula um percentil específico
# Aqui usamos o percentil 95
# Muito usado em análises clínicas e epidemiológicas
print(
    "Percentil 95:",
    np.percentile(pressao_sistolica, 95)
)

print("-" * 40)

Média da pressão: 135.0
Mediana: 127.5
Desvio padrão: 21.602468994692867
Variância: 466.6666666666667
Menor pressão: 115
Maior pressão: 180
Soma total das pressões: 810
Percentil 95: 170.0
----------------------------------------


## 📦 Geração de Dados (Mock Data)

Imagine que você está desenhando um algoritmo para monitorar a frequência cardíaca de pacientes em uma UTI. Você não pode simplesmente plugar seu código em desenvolvimento no sistema do hospital; um erro no código poderia causar falhas críticas.

O NumPy permite que você crie um "paciente virtual" ou uma "população simulada" instantaneamente para testar sua lógica com segurança.

As principais ferramentas de geração são:

* `np.zeros()`: Cria um bloco de memória preenchido por zeros (útil para inicializar contadores ou reservar espaço para exames).
* `np.ones()`: Cria um bloco preenchido pelo número 1 (útil para criar fatores de multiplicação ou estados padrão).
* `np.arange()`: Cria sequências numéricas ordenadas (perfeito para simular passagens de tempo, como dias de tratamento ou horas de incubação).
* `np.random`: O submódulo de aleatoriedade, ideal para simular flutuações biológicas reais (como batimentos cardíacos ou dosagens).

---

In [28]:
# np.zeros()
# Cria um array preenchido com zeros
# Muito usado para:
# - inicializar contadores
# - reservar memória
# - criar estruturas vazias

contadores = np.zeros(10)

print(
    "Array com zeros:",
    contadores
)

Array com zeros: [0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]


In [26]:
# np.ones()
# Cria um array preenchido com o valor 1
# Útil para:
# - fatores multiplicativos
# - máscaras
# - estados padrão

fatores = np.ones(5)

print(
    "Array com uns:",
    fatores
)

Array com uns: [1. 1. 1. 1. 1.]


In [27]:
# np.arange()
# Cria uma sequência numérica automática
# Funciona parecido com range()

# np.arange(início, fim, passo)

sequencia = np.arange(
    0,   # começa em 0
    10,  # termina antes do 10
    1    # incremento
)

print(
    "Sequência numérica:",
    sequencia
)

Sequência numérica: [0 1 2 3 4 5 6 7 8 9]


In [29]:

sequencia2 = np.arange(
    10,   # começa em 10
    1000,  # termina antes do 1000
    150    # incremento 
)

print(
    "Sequência numérica 2:",
    sequencia2
)


Sequência numérica 2: [ 10 160 310 460 610 760 910]


In [30]:
# -----------------------------------
# DADOS ALEATÓRIOS (SIMULAÇÃO)
# -----------------------------------

# np.random.randint()
# Gera números inteiros aleatórios

# Simulando batimentos cardíacos
# entre 60 e 100 bpm

batimentos = np.random.randint(
    60,   # mínimo
    101,  # máximo +1
    10    # quantidade de valores
)

print(
    "Batimentos cardíacos:",
    batimentos
)

Batimentos cardíacos: [71 78 98 95 69 75 92 63 63 75]


In [3]:
# np.random.normal()
# Gera dados seguindo distribuição normal
# Muito usado em:
# - bioestatística
# - simulações clínicas
# - variabilidade biológica

# Simulando glicemias próximas de 100

glicemia = np.random.normal(
    100,  # média
    15,   # desvio padrão
    5     # quantidade
)

print(
    "Glicemias simuladas:",
    glicemia
)

Glicemias simuladas: [106.94667283  96.45204306 103.08063571 100.52494289  96.63094786]


In [4]:
# np.round()
# Arredonda os números decimais

# round(array, 1)
# significa:
# - manter 1 casa decimal

glicemia = np.round(
    glicemia,
    1
)

# Exibe os valores simulados
print(
    "Glicemias simuladas:",
    glicemia
)

Glicemias simuladas: [106.9  96.5 103.1 100.5  96.6]
